CREATE_YOUR_FIRST_EXTENSION.md

```diff
@@ -147,53 +147,74 @@ public class CRUDExtension extends BaseExtension implements ActionExtension {
+    public List<NamedRoute> routes() {
         return List.of(
-            new RouteHandler(Method.PUT, "/sample", createHandler),
-            new RouteHandler(Method.GET, "/sample/{id}", readHandler),
-            new RouteHandler(Method.POST, "/sample/{id}", updateHandler),
-            new RouteHandler(Method.DELETE, "/sample/{id}", deleteHandler)
+            new NamedRoute.Builder().method(Method.PUT)
+                .path("/sample")
+                .uniqueName("crud_extension:sample/create")
+                .handler(createHandler)
+                .build(),
+            new NamedRoute.Builder().method(Method.GET)
+                .path("/sample/{id}")
+                .uniqueName("crud_extension:sample/get")
+                .handler(readHandler)
+                .build(),
+            new NamedRoute.Builder().method(Method.POST)
+                .path("/sample/{id}")
+                .uniqueName("crud_extension:sample/post")
+                .handler(updateHandler)
+                .build(),
+            new NamedRoute.Builder().method(Method.DELETE)
+                .path("/sample/{id}")
+                .uniqueName("crud_extension:sample/delete")
+                .handler(deleteHandler)
+                .build()
         );
     }
 
-    Function<RestRequest, ExtensionRestResponse> createHandler = (request) -> {
+    Function<RestRequest, RestResponse> createHandler = (request) -> {
         return new ExtensionRestResponse(request, RestStatus.OK, "To be implemented");
     };
 
-    Function<RestRequest, ExtensionRestResponse> readHandler = (request) -> {
+    Function<RestRequest, RestResponse> readHandler = (request) -> {
         return new ExtensionRestResponse(request, RestStatus.OK, "To be implemented");
     };
 
-    Function<RestRequest, ExtensionRestResponse> updateHandler = (request) -> {
+    Function<RestRequest, RestResponse> updateHandler = (request) -> {
         return new ExtensionRestResponse(request, RestStatus.OK, "To be implemented");
     };
 
-    Function<RestRequest, ExtensionRestResponse> deleteHandler = (request) -> {
+    Function<RestRequest, RestResponse> deleteHandler = (request) -> {
         return new ExtensionRestResponse(request, RestStatus.OK, "To be implemented");
     };
 }
 ```
 
 ## Use OpenSearch clients to implement functionality
 
-During the initial creation of the extension, you either implemented `setExtensionsRunner()` yourself or used the `BaseExtension` class, which does it for you. This gives you access to the `ExtensionsRunner` object that is running this extension. The `ExtensionsRunner` has getters that provide access to many objects you will need, one of which is the `SDKClient`. The `SDKClient` class allows initialization of the OpenSearch Java client, which has synchronous and asynchronous versions. For simplicity, this example uses the synchronous client.
+To use the OpenSearch REST API, you will need an instance of the OpenSearch Java Client.
```

Comment:
SDKClient as a wrapper of Java Client?

Reply:
Not sure what the question is?  

SDKClient doesn't wrap Java Client, it just gives a one-liner initialization option to a non-secure client, useful for tutorials like this.

Reply:
Should we mention? 
```suggestion
To use the OpenSearch REST API, you will need an instance of the SDKClient
```

Reply:
No, we do not need the SDKClient.  

Users can create their own OpenSearchClient. I linked to two options that are in the OpenSearch Client documentation and that is the preferred way to do it, particularly if the users need TLS/SSL.

Reply:
On line 217, it's mentioned 
``SDKClient` class allows initialization of the OpenSearch Java client`

Reply:
> Can we add java client and SDKClient as 2 options then?

I thought that's exactly what I'm doing here.  I cite the Java Client docs and then I explain how to do it (currently) with SDKClient.

> # Use OpenSearch clients to implement functionality
> 
> To use the OpenSearch REST API, you will need an instance of the OpenSearch Java Client.
> 
> If you require SSL and TLS capability, refer to the OpenSearch Java Client documentation for either [Apache HttpClient 5 Transport](https://opensearch.org/docs/latest/clients/java/#initializing-the-client-with-ssl-and-tls-enabled-using-apache-httpclient-5-transport) or [OpenSearch RestClient Transport](https://opensearch.org/docs/latest/clients/java/#initializing-the-client-with-ssl-and-tls-enabled-using-restclient-transport).
> 
> The `SDKClient` class allows initialization of the OpenSearch Java client ...

Now I may be wrong or imprecise about SSL/TLS.  Whatever we're doing is not secure.

Personally I'd like to not mention SDKClient in this doc at all.  Point them to the official docs for the Java client and don't try to include that in this sample.

Reply:
I've updated this to not even mention SDKClient or how to get a copy of it.  We essentially implement a non-secure version of what's on the docs and we shouldn't point people to that non-production code.  Further, I'm arguing elsewhere (#612, #625) that we should remove it.

That section now reads:

> ## Use OpenSearch Java Client to implement functionality
> 
> To use the OpenSearch REST API, you will need an instance of the OpenSearch Java Client.
> 
> Refer to the OpenSearch Java Client documentation for either [Apache HttpClient 5 Transport](https://opensearch.org/docs/latest/clients/java/#initializing-the-client-with-ssl-and-tls-enabled-using-apache-httpclient-5-transport) or [OpenSearch RestClient Transport](https://opensearch.org/docs/latest/clients/java/#initializing-the-client-with-ssl-and-tls-enabled-using-restclient-transport).
> 
> The remainder of this example assumes you have implemented one of the above options in your constructor:
> 
> ```java
> private final OpenSearchClient client;
> 
> public CrudAction() {
>     final OpenSearchTransport transport = // implement per documentation
>     this.client = new OpenSearchClient(transport);
> }
> ```